Arrow 23.0.0

thisisnic opened the pull request:
Closed #47956 as rebase was messy.  mimalloc is now added to dependencies.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

dg0yt commented:
FTR the vendored mimalloc is version 3. 
Trying to update the vcpkg port to this version showed run-time issues on systems with 16 KiB page size (arm osx, potentially arm android) when feature `secure` is enabled: #44699
Reported upstream three weeks ago, but not response at all: https://github.com/microsoft/mimalloc/issues/1166

dg0yt commented:
> @dg0yt - I took a look to see if it makes sense for us to use the current version of mimalloc from vcpkg, but it appears not as Arrow requires a newer version of mimalloc, >= 3.0.9, due to significant memory leak issues with older versions ([apache/arrow#47266](https://github.com/apache/arrow/issues/47266)) and recommendations in [microsoft/mimalloc#1073](https://github.com/microsoft/mimalloc/issues/1073) to use v3 and above for production usage.

AFAIU the links show evidence that there is a problem with mimalloc 3.0. And that version was never marked as recommended version.
There is no information about problems with the current version of mimalloc in vcpkg, 2.2.
mimalloc 3.2 seems to become the recommended version for production just now, and a PR is getting ready for updating the vcpkg port, #44699. (Delayed by upstream and by vcpkg CI. And still labeled "release candidate")

thisisnic committed:
[arrow] Update to 23.0.0

- Use vcpkg mimalloc instead of building from source (Conan approach)
- Add mimalloc as dependency

thisisnic committed:
Update versions

thisisnic committed:
Remove redundant patch

thisisnic committed:
add alias

thisisnic committed:
Fix mimalloc patch line count

thisisnic committed:
Remove arrow[orc] from feature-fails baseline (now passes on Windows)

BillyONeal reviewed (approved):
Thanks for devendoring!

dg0yt commented on ports/arrow/0007-use-vcpkg-mimalloc.patch:
```diff
@@ -0,0 +1,17 @@
++  else()
++    add_library(mimalloc::mimalloc ALIAS mimalloc)
++  endif()
++endif()
```
The patch is here, so start in this repo? Basically a proper error report will give a hint where it belongs to, and maybe answer other questions,
